docs/tutorials/python/float16.md

```diff
@@ -0,0 +1,280 @@
+
+### For accuracy
+
+#### Multi precision mode
+When training in float16, it is advisable to still store the master copy of the weights in float32 for better accuracy. The higher precision of float32 helps overcome cases where gradient update can become 0 if represented in float16. This mode can be activated by setting the parameter `multi_precision` of optimizer params to `True` as in the above example. It has been found that this is not required for all networks to achieve the same accuracy as with float32, but nevertheless recommended. Note that for distributed training, this is currently slightly slower than without `multi_precision`, but still faster than using float32 for training.
+
+#### Large reductions 
+Since float16 has low precision for large numbers, it is best to leave layers which perform large reductions in float32. This includes BatchNorm and Softmax. Ensuring that batchnorm performs reduction in float32 is handled by default in both Gluon and Module APIs. While Softmax is set to use float32 even during float16 training in Gluon, in the Module API there needs to be a cast to float32 before softmax as the above symbolic example code shows.
+
+#### Loss scaling
+For some networks just switching the training to float16 mode was not found to be enough to reach the same accuracy as when training with float32. This is because the activation gradients computed are too small and could not be represented in float16 representable range. Such networks can be made to achieve the accuracy reached by float32 with a couple of changes. 
+
+Most of the float16 representable range is not used by activation gradients generally. So we can shift the gradients into float16 range by scaling up the loss by a factor `S`. Essentially we scale up the loss before backward pass, and then scale back the gradients before updating the weights.
+
+Here's how you can configure the loss to be scaled up by 128 and rescale the gradient down before updating the weights.
+
+*Gluon*
+```
+loss = gluon.loss.SoftmaxCrossEntropyLoss(weight=128)
+optimizer = mx.optimizer.create('sgd', multi_precision=True, rescale_grad=1.0/128)
+```
+*Module*
+```
+mxnet.sym.SoftmaxOutput(other_args, grad_scale=128.0)
+optimizer = mx.optimizer.create('sgd', multi_precision=True, rescale_grad=1.0/128)
+```
+
+Networks like Multibox SSD, R-CNN, bigLSTM and Seq2seq were found to exhibit such behavior. Refer the linked articles below for more details on this.
+    
+## References
+1. [Training with Mixed Precision User Guide](http://docs.nvidia.com/deeplearning/sdk/mixed-precision-training/index.html)
+2. [Mixed Precision Training at ICLR 2018](https://arxiv.org/pdf/1710.03740.pdf)
+3. [Mixed-Precision Training of Deep Neural Networks](https://devblogs.nvidia.com/mixed-precision-training-deep-neural-networks/)
+
```

Comment:
Can you add this `<!-- INSERT SOURCE DOWNLOAD BUTTONS -->` at the end of your .md file to enable the notebook download of your tutorial? Thanks!